### PR TITLE
Fft 301-create user

### PR DESCRIPTION
**Pull request by adnaniroshanak**

### CODE REVIEW

File: api.swagger.yaml [hunk 1]
```diff
@@ -948,6 +948,47 @@ paths:
...
+        - "application/json"
+      parameters:
+        - in: "body"
+          name: "body"
+          description: "User object"
+          required: true
+          schema:
+            $ref: "#/definitions/UserForCreation"
+      responses:
+        200:
```
Comment (timdauer) [inline api.swagger.yaml:975]:
```suggestion
        201:
```

File: api.swagger.yaml [hunk 2]
```diff
@@ -948,6 +948,47 @@ paths:
...
+          description: "The user is successfully created."
```
Comment (timdauer) [inline api.swagger.yaml:976]:
```suggestion
          description: "The user was successfully created."
```

File: api.swagger.yaml [hunk 3]
```diff
@@ -2271,3 +2312,64 @@ definitions:
...
         type: "number"
         example: "30"
+  UserForCreation:
+    type: "object"
+    required:
+      - "username"
+      - "password"
+      - "roles"
+    properties:
+      userId:
```
Comment (timdauer) [inline api.swagger.yaml:2322]: This field should not be supplied during creation - it is generated when we register the user with google identity toolkit

File: api.swagger.yaml [hunk 4]
```diff
@@ -2271,3 +2312,64 @@ definitions:
...
+      username:
+        type: "string"
+        example: "aschneider"
+      password:
+        type: "string"
+        example: "fsdf6556"
+      roles:
+        type: "array"
+        description: "roles of a user "
+        minItems: 1
```
Comment (timdauer) [inline api.swagger.yaml:2341]:
```suggestion
        minItems: 1
        maxItems: 1
```

File: api.swagger.yaml [hunk 5]
```diff
@@ -2271,3 +2312,64 @@ definitions:
...
+          type: "string"
+          enum:
+            - "FULFILLER"
+            - "SUPERVISOR"
+            - "ADMINISTRATOR"
+    xml:
+      name: "UserForCreation"
+  User:
+    allOf:
+      - $ref: '#/definitions/VersionedResource'
```
Comment (timdauer) [inline api.swagger.yaml:2352]:
```suggestion
      - $ref: '#/definitions/VersionedResource'
      - $ref: '#/definitions/UserForCreation'
```

File: api.swagger.yaml [hunk 6]
```diff
@@ -2271,3 +2312,64 @@ definitions:
...
+      - required:
+          - "username"
+        type: "object"
+        properties:
```
Comment (timdauer) [inline api.swagger.yaml:2356]: If you extend the user for creation and the versioned resource I think you do not need any of the following fields

File: api.swagger.yaml (hunk 3 above)
Comment (adnaniroshanak) [reply api.swagger.yaml]: fixed

File: api.swagger.yaml (hunk 5 above)
Comment (MarkusWallnerFul) [reply api.swagger.yaml:2349]: But if we extend the user with allOf user for creation, we would add the password to our internal user model. And i do not think that we want this

**Pull request merged**